pkgs/development/python-modules/gpt4all-bindings/default.nix

```
@@ -0,0 +1,77 @@
+{
+  lib,
+  gpt4all,
+  gpt4all-cuda,
+  buildPythonPackage,
+  stdenv,
+  requests,
+  tqdm,
+  pythonOlder,
+  setuptools,
+  wheel,
+  python,
+  cuda ? false,
+}:
+let
+  basepkg = if cuda then gpt4all-cuda else gpt4all;
+
+  gpt4all-backend = stdenv.mkDerivation {
+    pname = "gpt4all-backend";
+    inherit (basepkg)
+      src
+      version
+      cmakeFlags
+      nativeBuildInputs
+      buildInputs
+      ;
+
+    sourceRoot = "${basepkg.src.name}/gpt4all-backend";
+    installPhase = ''
+      cp -r ./ $out
+    '';
+  };
+in
+buildPythonPackage {
+  pname = "gpt4all-bindings";
+  inherit (basepkg) src version;
+  sourceRoot = "${basepkg.src.name}/gpt4all-bindings/python";
+  pyproject = false;
+
+  patches = [ ./setup_fixup.patch ];
+  disabled = pythonOlder "3.12";
```

```suggestion
  disabled = pythonOlder "3.8";
```
or remove this line.
https://github.com/nomic-ai/gpt4all/blob/a2b4529945460b200a226bc22f4861c24d285049/gpt4all-bindings/python/setup.py#L88

---

I have a build failure on python older than 3.12, some syntax error somewhere in their code.
I'm not sure where it comes from yet, it may simply be that they write code for 3.12 and forgot to bump the requirement in `setup.py`.